### PR TITLE
chore(docs): Update Gatsby framework version support information

### PR DESCRIPTION
**Pull request by pragmaticpat**
## Description

Beginning to add our planned maintenance schedule for Gatsby 4, Gatsby 3. I'm happy to adjust the layout / content as needed to help it better align with the rest of our docs.

I ultimately would like the page in this PR to supersede our existing v2 maintenance doc so we don't have version-specific maintenance pages.

- [x] Remove unrelated commits from this PR!
- [x] Update the readme to point to this new page
- [x] Delete the former page ( v2-maintenance-plan )
- [x] Perform sidebar updates in .com
- [x] Add redirects to .com

This change needs to be coordinated with .com PR 13419

[ch40162]

### CODE REVIEW

**Comment by LekoArts** [comment]
There are unrelated changes in this PR :)

I think we also should add a link to this then in the README (both in root and in packages/gatsby/README.md). We also have this doc still: https://www.gatsbyjs.com/docs/reference/release-notes/v2-maintenance-plan/

We'll need to delete it https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/reference/release-notes/v2-maintenance-plan.md and add a redirect to the new doc

**Comment by pragmaticpat** [comment]
🤦 - I'll rip those unrelated commits out!

Good call on the link in the readme

And yes, we need to delete the previous page when this one gets published, per my comment when I opened the PR:

>I ultimately would like the page in this PR to supersede our existing v2 maintenance doc so we don't have version-specific maintenance pages.

**Comment by pragmaticpat** [comment]
@LekoArts - I'm trying a more concise version of the "supported versions" page. Let me know what you think!

File: README.md [hunk 1]
```diff
@@ -147,6 +147,8 @@ We welcome suggestions for improving our docs. See the [“how to contribute”]
 
 Wondering what we've shipped recently? Check out our [release notes](https://www.gatsbyjs.com/docs/reference/release-notes) for key highlights, performance improvements, new features, and notable bugfixes.
 
+Also, [see here](https://www.gatsbyjs.com/docs/reference/release-notes/gatsby-version-support/) to understand the version support plans for each version of Gatsby.
```
Comment (LekoArts) [inline README.md:150]:
```suggestion
Also, read our [documentation on version support](https://www.gatsbyjs.com/docs/reference/release-notes/gatsby-version-support/) to understand our plans for each version of Gatsby.
```

Better link text than "see here"

File: docs/docs/reference/release-notes/gatsby-version-support.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+title: Gatsby Framework Version Support
+---
+
+## Introduction
+
+This page illustrates the support plans for the major versionf of Gastby. This allows you to effectively plan their Gatsby framework upgrades.
```
Comment (LekoArts) [inline docs/docs/reference/release-notes/gatsby-version-support.md:7]:
```suggestion
This page illustrates the support plans for the major versions of Gatsby. This allows you to effectively plan your Gatsby framework upgrades.
```

File: docs/docs/reference/release-notes/gatsby-version-support.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+| Version | Status                        | As Of            | Until           |
+| ------- | ----------------------------- | ---------------- | --------------- |
+| 4       | Active                        | October 21, 2021 | Summer 2022     |
+| 3       | Maintenance Long-term support | October 21, 2021 | Summer 2022     |
+| 2       | Maintenance Long-term support | March 3, 2021    | January 1, 2022 |
+
+## Terminology
+
+- **Critical Patch** - A critical patche addresses an issue that prevents the Gatsby framework from delivering intended user value under typical conditions. Other items, such as minor bug fixes and new features are not considered "critical patches" to Gatsby.
```
Comment (LekoArts) [inline docs/docs/reference/release-notes/gatsby-version-support.md:23]:
```suggestion
- **Critical Patch** - A critical patch addresses an issue that prevents the Gatsby framework from delivering intended user value under typical conditions. Other items, such as minor bug fixes and new features are not considered "critical patches" to Gatsby.
```

**Review by LekoArts — APPROVED** [review]

**Pull request merged**